src/main/java/org/opensearch/cluster/etcd/ETCDStateDeserializer.java

```diff
@@ -155,7 +169,8 @@ private static CoordinatorNodeState readCoordinatorNodeState(DiscoveryNode local
         Map<Index, List<List<NodeShardAssignment>>> remoteShardAssignment = new HashMap<>();
         for (Map.Entry<String, Object> indexEntry : indices.entrySet()) {
             Map<String, Object> indexConfig = (Map<String, Object>) indexEntry.getValue();
-            String uuid = (String) indexConfig.get("uuid");
+            // Generate UUID programmatically instead of reading from etcd
+            String uuid = generateDeterministicUUID(indexEntry.getKey());
```

Comment:
Instead of `generateDeterministicUUID`,what if we just say `String uuid = (String) indexConfig.getOrDefault("uuid", indexEntry.getKey())`?

The index name should be consistent across all nodes.